tests/test_events/test_api.py

```diff
@@ -199,15 +199,31 @@ def test_singleton_pattern_working_correctly_for_event_api(self, mock_send: Magi
         )
 
     @patch.object(Boto3Client, "eventbridge", MagicMock())
-    def test_second_send_clears_everything(self) -> None:
+    def test_second_send_does_not_clear_everything(self) -> None:
         event = MyTestEvent({"x": 1})
         self.event_api.register(event)
 
         self.event_api.send()
         self.event_api.send()
 
         assert self.event_api.failed_events == []
-        assert self.event_api.sent_events == []
+        assert self.event_api.sent_events == [event]
+        assert self.event_api.pending_events == []
+
+    @patch.object(Boto3Client, "eventbridge", MagicMock())
+    def test_conclusive_send_extends_event_list(self) -> None:
```

Comment:
It would be good to test it also with the failed events as the second `.send()`, even if fully correct, will raise an error.

I am not 100% sure about this logic but that is how behaves after your changes 🤔

Reply:
You skipped the above comment ☝️